File: platforms/android/devkit/devkit/src/main/java/io/nativeblocks/devkit/session/RemoteSessionImpl.kt
package io.nativeblocks.devkit.session

import io.nativeblocks.devkit.lib.cache.INativeCache

internal class RemoteSessionImpl(private val cache: INativeCache) : IRemoteSession {
    override fun getEndpoint(): String {
        return cache.pull("ENDPOINT", "")
    }

    override fun getToken(): String {
        return cache.pull("AUTH_TOKEN", "")
    }

    override fun getRealtimeEndpoint(): String {
        return cache.pull("REALTIME_ENDPOINT", "")
    }

    override fun getUsername(): String {
        return cache.pull("USERNAME", "")
    }

    override fun getUUID(): String {
        return cache.pull("UUID", "")
    }
}

File: platforms/ios/devkit/Sources/NativeblocksDevkit/session/RemoteSessionImpl.swift
import Foundation

internal class RemoteSessionImpl: IRemoteSession {
    private let cache: INativeCache

    init(cache: INativeCache) {
        self.cache = cache
    }

    func getEndpoint() -> String {
        return cache.pull(key: "ENDPOINT", defaultValue: "")
    }

    func getToken() -> String {
        return cache.pull(key: "AUTH_TOKEN", defaultValue: "")
    }

    func getRealtimeEndpoint() -> String {
        return cache.pull(key: "REALTIME_ENDPOINT", defaultValue: "")
    }

    func getUsername() -> String {
        return cache.pull(key: "USERNAME", defaultValue: "")
    }

    func getUUID() -> String {
        return cache.pull(key: "UUID", defaultValue: "")
    }
}
